ToolChain/Assembler/Parser/Parser.py

```diff
@@ -381,91 +355,41 @@ def translateRegisterNameToRegisterCode(self, registerName: str=""):
-        elif registerName == "E":
-            registerCode = REGISTER_E
-        elif registerName == "F":
-            registerCode = REGISTER_F
-        elif registerName == "G":
-            registerCode = REGISTER_G
-        elif registerName == "G":
-            registerCode = REGISTER_G
-        elif registerName == "A2":
-            registerCode = REGISTER_A2
-        elif registerName == "B2":
-            registerCode = REGISTER_B2
-        elif registerName == "C2":
-            registerCode = REGISTER_C2
-        elif registerName == "D2":
-            registerCode = REGISTER_D2
-        elif registerName == "E2":
-            registerCode = REGISTER_E2
-        elif registerName == "F2":
-            registerCode = REGISTER_F2
-        elif registerName == "G2":
-            registerCode = REGISTER_G2
-        elif registerName == "S":
-            registerCode = REGISTER_S
-        elif registerName == "S2":
-            registerCode = REGISTER_S2
-        else:
-            raise ValueError("Invalid register given. '{}' seen as input, expect valid register".format(registerName))
+
+        try:
+            registerCode = REGISTERS[registerName.upper()]
+
+        except KeyError as e:
+            raise ValueError("Invalid instruction format: Too many arguments at line " + str(self.lineno))
 
         return registerCode
 
-    def translateTextImmediateToImmediate(self, textImmediate: str=""):
+    def translateTextImmediate(self, textImmediate: str=""):
         """
         This will translate an immediate value in a way that can be understood by the architecture.
         :param textImmediate: str, an immediate value to be translated
         :return: int, an immediate that can be worked on
         """
-        immediate = None
-        isNegative = False
-        textImmediate = textImmediate.lower()  # Needed in case of 0XFF instead of 0xFF
-
-        if textImmediate[0] == "-":
-            isNegative = True
-            textImmediate = textImmediate[1:]
-
-        if len(textImmediate) > 2 and textImmediate[0:2] == "0b":
-            # Indicates binary immediate
-            baseToUse = 2
-            textImmediate = textImmediate[2:]
-        elif len(textImmediate) > 2 and textImmediate[0:2] == "0x":
-            # Indicate hexadecimal immediate
-            baseToUse = 16
-            textImmediate = textImmediate[2:]
-        else:
-            # Take a leap of faith! This should be base 10
-            baseToUse = 10
 
-        immediate = int(textImmediate, baseToUse)
+        textImmediate = textImmediate.lower()  # Needed in case of 0XFF instead of 0xFF
+        immediate = int(textImmediate, 0)
 
-        validationImmediate = immediate
-        immediate &= 0xFFFFFFFF  # Maximum immediate value is 32 bits
+        if immediate > 0xFFFFFFFF or immediate < -0xFFFFFFFF:
+            # Maximum value is 0xFFFFFFFF
+            raise ValueError("Given immediate value is too big, {} received but "
```

Comment:
Good catch!

Reply:
This is actually wrong. 
We have 32 bits registers. The sign of the numbers is managed in the core as a two's complement. 
Therefore, -0xFFFFFFFF is twice as small as the smaller number that the core would be able to handle. The previous code (while could require more testing?) was enforcing 32 bits numbers while the suggested code does not do this enforcement. Positiveness or negativeness of numbers is to be manually handled as two's complement because of how python handles the INT type.